block/manager.go

```diff
@@ -643,14 +671,17 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 		// set the commit to current block's signed header
 		block.SignedHeader.Commit = *commit
 
-		// SaveBlock commits the DB tx
-		err = m.store.SaveBlock(block, commit)
-		if err != nil {
-			return err
-		}
 	}
-
-	// Apply the block but DONT commit
+	vals, err := m.getValsForCentralizedSequencer()
```

Comment:
why do this on every block? can you do it once in `NewFromGenesisDoc` and store `vals` in the manager for reuse?